src/Flow.jl

```diff
@@ -59,17 +59,21 @@ lowerBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop (
     Φ[I] = ϕuP(j,CIj(j,CI(I,i),N[j]-2),CI(I,i),u,ϕ(i,CI(I,j),u)) -ν*∂(j,CI(I,i),u); r[I,i] += Φ[I]) over I ∈ slice(N,2,j,2)
 upperBoundary!(r,u,Φ,ν,i,j,N,::Val{true}) = @loop r[I-δ(j,I),i] -= Φ[CIj(j,I,2)] over I ∈ slice(N,N[j],j,2)
 
-using EllipsisNotation
+using EllipsisNotation #, Setfield
 """
     accelerate!(r,t,g)
 
 This function adds a uniform acceleration field `g` at time `t` to `r`.
 If `g ≠ nothing`, then `g(i,t)=dUᵢ/dt`.
 """
-accelerate!(r,t,g) = for i ∈ 1:last(size(r))
+accelerate!(U,r,t,g,::Nothing) = for i ∈ 1:last(size(r))
     r[..,i] .+= g(i,t)
 end
-accelerate!(r,t,::Nothing) = nothing
+accelerate!(U,r,t,::Nothing,Uₜ::Function) = for i ∈ 1:last(size(r))
+    r[..,i] .+= ForwardDiff.derivative(τ->Uₜ(i,τ),t)
+    U[i] = Uₜ(i,t)
```

Comment:
This line is causing the trouble; it works fine on the CPU, but the GPU complains.